### PR TITLE
Validations

### PR DESCRIPTION
**Pull request by MayPyone**
-Write data validations for models
-Add unit tests

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+RSpec.describe Post, type: :model do
+  let(:user) { User.new(name: 'user1') }
+
+  subject { Post.new(title: nil, text: 'Hi', comment_counter: 0, like_counter: 2) }
+
+  it 'is valid with valid attributes' do
+    expect(subject).to be_present
+  end
+
```
Comment (1GORDON) [inline spec/models/post_spec.rb:13]: Great job @MayPyone 🤝
- As stated in the project requirements, kindly ensure the missing post-model validation tests are added. These include `CommentsCounter must be an integer greater than or equal to zero.`, and `LikesCounter must be an integer greater than or equal to zero`.
post-model validations

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi @MayPyone ✋

Good job so far!

## Highlights!
- All tests are passing ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
- Kindly update the sections of your readme which are referencing a different project `Catalog-of-my-things` instead of `Blog`.
catalog

- Kindly update the issues link. The issues referenced by the link are for a different project `Catalog-of-my-things` instead of `Blog`.
issues

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb [hunk 2]
```diff
@@ -2,7 +2,8 @@ class User < ApplicationRecord
   has_many :posts, foreign_key: 'author_id'
   has_many :likes
   has_many :comments
-
+  validates :name, presence: true
+  validates :post_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
   def recent_posts
     posts.order(created_at: :desc).limit(3)
   end
```
Comment (franclobo) [inline app/models/user.rb:9]: - **[OPTIONAL]** Please 🙏🏽 add unit specs for all of your models' methods and validations to complete the milestone requirements. 👇🏽 

> Add unit specs for all of your models' methods and validations.

File: app/models/post.rb [hunk 3]
```diff
@@ -3,19 +3,14 @@ class Post < ApplicationRecord
   has_many :likes
   has_many :comments
   after_save :update_posts_counter
+  validates :title, presence: true, length: { maximum: 250 }
+  validates :comment_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+  validates :like_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
   def recent_comments
     comments.order(created_at: :desc).limit(5)
   end
```
Comment (franclobo) [inline app/models/post.rb:11]: - **[OPTIONAL]** Please 🙏🏽 add unit specs for all of your models' methods and validations to complete the milestone requirements. 👇🏽 

> Add unit specs for all of your models' methods and validations.

**Review by franclobo — APPROVED** [review]
Hi @MayPyone, 

# STATUS: APPROVED 😃 

git-merge

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️ 
- All tests passed ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**